src/battle_util.c

```diff
@@ -2021,18 +2017,15 @@ u8 AbilityBattleEffects(u8 caseID, u8 bank, u8 ability, u8 special, u16 moveArg)
             case ABILITY_CLOUD_NINE:
             case ABILITY_AIR_LOCK:
                 {
-                    u8 i;
-
-                    for (i = 0; i < gNoOfAllBanks; i++)
+                    // that's a weird choice for a variable, why not use i or bank?
+                    for (target1 = 0; target1 < gNoOfAllBanks; target1++)
                     {
-                        // TODO: i should be in r6 here
-                        //asm("":::"r4","r5");
-                        effect = CastformDataTypeChange(i);
+                        effect = CastformDataTypeChange(target1);
                         if (effect != 0)
                         {
                             BattleScriptPushCursorAndCallback(BattleScript_CastformChange);
-                            gBattleScripting.bank = i;
-                            gBattleStruct->formToChangeInto = effect - 1;
+                            gBattleScripting.bank = target1;
+                            *(&gBattleStruct->formToChangeInto) = effect - 1;
```

Comment:
Is making it into a pointer and dereferencing necessary?

Reply:
Just check out other battle files. It's a leftover from R/S direct ewram usage and everything being a u8 array;